src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -76,16 +76,41 @@ if (__DEV__) {
     validateUnknownProperties(type, props);
   };
 
-  var warnForTextDifference = function(serverText: string, clientText: string) {
+  // HTML parsing normalizes CR and CRLF to LF.
+  // It also can turn \u0000 into \uFFFD inside attributes.
+  // https://www.w3.org/TR/html5/single-page.html#preprocessing-the-input-stream
+  // If we have a mismatch, it might be caused by that.
+  // We won't be patching up in this case as that matches our past behavior.
+  var NORMALIZE_NEWLINES_REGEX = /\r\n?/g;
+  var NORMALIZE_NULL_AND_REPLACEMENT_REGEX = /\u0000|\uFFFD/g;
+
+  var normalizeMarkupForTextOrAttribute = function(markup: mixed): string {
+    const markupString = typeof markup === 'string'
+      ? markup
+      : '' + (markup: any);
```

Comment:
Not super proud but I couldn't get Flow to understand I'm just trying to cast to string.

Reply:
`String(markup)`